### ISSUE TITLE
Menu popup position beneath button/icon?

### ISSUE THREAD

**Issue by cstrat** [mantinedev/mantine#16]
@mantine/core - Menu

See example here where I've made it work with `menuPosition` prop.
https://codesandbox.io/s/mantine-menu-popup-position-8xegv

However I feel this is a bit hacky and it might be better if the `Menu` component were able to set a `basePosition` somewhere else.

For example if you set the `basePosition` of the Menu to 'bottom-left' then the 0:0 coordinate of the Menu is the bottom-left of the control component. I am sure you get my suggestion, this would still allow for adjustments via `menuPosition` where you can nudge it with left/right/top/bottom values...

Any thoughts on this one? I am happy to just keep doing what I've got in that demo - but having the Menu obscure the button is not the best in my application.

**Comment by rtivital** [comment]
I think it's a great idea to do this, but two props for position is no-go – it breaks the single source of truth principle and will definitely bring unnecessary confusion. It shall be implemented something like this to support both options via single prop:

```tsx
interface MenuProps {
  menuPosition?:
    | {
        top?: React.CSSProperties['top'];
        bottom?: React.CSSProperties['bottom'];
        left?: React.CSSProperties['left'];
        right?: React.CSSProperties['right'];
      }
    | {
        position: 'top' | 'left' | 'right' | 'bottom';
        placement: 'start' | 'center' | 'end';
        gutter?: number;
      };
}

// Same as tooltip
<Menu menuPosition={{ position: 'left', placement: 'start' }} />

// Still supports initial variant
<Menu menuPosition={{ top: 10, left: 10 }} />
``` 

Can you try out this and open a PR if this works out?

**Comment by cstrat** [comment]
> I think it's a great idea to do this, but two props for position is no-go
I agree, I only suggested that because I didn't want to suggest a breaking change, but what you're saying makes more sense to support both options with the one prop.

> Can you try out this and open a PR if this works out?
I've never used Typescript before but happy to give it a try later today if I have a chance!

**Comment by cstrat** [comment]
That was a learning experience!

I got it all running following the contribution guideline.

After editing the storybook entry with the modified code and it all works fine.
I just couldn't work out how to add some spacing so that it doesn't sit directly against the button but `gutter` doesn't appear to do anything.

![image](https://user-images.githubusercontent.com/4209076/117238651-a6075280-ae70-11eb-8b63-43f7c6f0311d.png)

With gutter value:

![image](https://user-images.githubusercontent.com/4209076/117238819-ecf54800-ae70-11eb-9248-6f1782aed10f.png)

**Comment by cstrat** [comment]
Ah ok it looks like with tooltip there are some styles associated which setting `gutter` applies.
I will keep looking into it :)

**Comment by rtivital** [comment]
Looks good, by the way, it's not a breaking change if we add extra option to `menuPosition` prop – it will still support all previous options

**Comment by cstrat** [comment]
It looks like there are a whole bunch of styles which are applied to tooltips which interact with the `placement` and `gutter` props. They would need to be replicated for the menu popup if those props were applied too.

Now I am sure I could hack together the code to make it work, but I already feel out of my depth with typescript and I've also never used `jss` for react styling. I will have a go at this but whatever I come up with won't really be the right quality for a PR. I can submit it anyway.

By the way, why isn't there a hosted Storybook? It is a great way to see all the components.

> it's not a breaking change if we add extra option to menuPosition prop – it will still support all previous options

Yeah I didn't think to do it the way you said 👍🏻

**Comment by rtivital** [comment]
I use storybook for local development only for visual testing, it almost never includes all the cases for components and some stories look ugly and I believe that all examples should show how component may be used in real application, not dummy placeholder lorem ipsum with only edge cases. So the storybook will remain local.

**Comment by rtivital** [comment]
Released in 2.3.0, menu position and placement now can be changed the same way as Popover
